profiler/src/main/java/com/splunk/opentelemetry/profiler/allocation/exporter/PprofAllocationEventExporter.java

```diff
@@ -0,0 +1,317 @@
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.zip.GZIPOutputStream;
+import jdk.jfr.consumer.RecordedEvent;
+import jdk.jfr.consumer.RecordedFrame;
+import jdk.jfr.consumer.RecordedMethod;
+import jdk.jfr.consumer.RecordedStackTrace;
+import jdk.jfr.consumer.RecordedThread;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PprofAllocationEventExporter implements AllocationEventExporter {
+  private static final Logger logger = LoggerFactory.getLogger(PprofAllocationEventExporter.class);
+
+  private final LogProcessor logProcessor;
+  private final Resource resource;
+  private final AllocationDataFormat allocationDataFormat;
+  private final EventPeriods eventPeriods;
+  private ProfileData profileData = new ProfileData();
+
+  private PprofAllocationEventExporter(Builder builder) {
+    this.logProcessor = builder.logProcessor;
+    this.resource = builder.resource;
+    this.allocationDataFormat = builder.allocationDataFormat;
+    this.eventPeriods = builder.eventPeriods;
+  }
+
+  @Override
+  public void export(RecordedEvent event, AllocationEventSampler sampler, SpanContext spanContext) {
+    RecordedStackTrace stackTrace = event.getStackTrace();
+    if (stackTrace == null) {
+      return;
+    }
+
+    Profile.Builder profile = profileData.profile;
+    LocationTable locationTable = profileData.locationTable;
+    StringTable stringTable = profileData.stringTable;
+
+    long allocationSize = event.getLong("allocationSize");
+
+    Sample.Builder sample = Sample.newBuilder();
+    sample.addValue(allocationSize);
+    // XXX StackSerializer limits stack depth. Although I believe jfr also limits it should we
+    // attempt to limit the depth here too?
+    for (RecordedFrame frame : stackTrace.getFrames()) {
+      RecordedMethod method = frame.getMethod();
+      if (method == null) {
+        sample.addLocationId(locationTable.get("", "unknown.unknown", -1));
+      } else {
+        sample.addLocationId(
+            locationTable.get(
+                "", // file name is not known
+                method.getType().getName() + "." + method.getName(),
+                frame.getLineNumber()));
+      }
+    }
+
+    String eventName = event.getEventType().getName();
+    addLabel(sample, SOURCE_EVENT_NAME.getKey(), event.getEventType().getName());
+    Duration eventPeriod = eventPeriods.getDuration(eventName);
+    if (!EventPeriods.UNKNOWN.equals(eventPeriod)) {
+      addLabel(sample, SOURCE_EVENT_PERIOD.getKey(), String.valueOf(eventPeriod.toMillis()));
+    }
+    Instant time = event.getStartTime();
+    addLabel(sample, "source.event.time", String.valueOf(time.toEpochMilli()));
+
+    RecordedThread thread = event.getThread();
+    addLabel(sample, "thread.id", String.valueOf(thread.getJavaThreadId()));
```

Comment:
On the backend we index native thread Id or `nid` from the thread dump metadata line, I presume it is from `thread.getOSThreadId()`.

Reply:
You can just add `nid=0x...`